File: c++/1486. XOR Operation in an Array/1486.cpp
class Solution{
public:
    int xorOperation(int n, int start){
        int x = 0; //here will be next result
        vector<int>v; //new vector of ints
        for (int i=0;i<n;i++) //for looping in len of array (n)
        {
            v.push_back(start+2*i); //adding our formula to each index
        }
        for (int i=0;i<v.size();i++) //for looping in vector v
            x=x^v[i]; //and doing XOR
        return x;
    }
};

File: python/1486. XOR Operation in an Array/1486.py
class Solution:
    def bitwise(self,arr): # Function for determine result of XOR
        XOR = arr[0]
        for i in range(1,len(arr)):
            XOR = XOR^arr[i] #we need to know what is XOR
        return XOR
    def xorOperation(self, n: int, start: int) -> int:
        nums = [0 for x in range(n)] #first we give nums of all zeros
        for i in range(n): #and later for all indexes we will fill values 
            nums[i] = start + 2*i
        return self.bitwise(nums) #then we XORing our nums
